Bad symbol is selected leading to ClassCastException. Regression from scala 2.13

## Compiler version

Scala 3.3.2 (3.4.0 also affected)

## Minimized code

https://scastie.scala-lang.org/YdGomTp2S0Sc48M47mXLEA
```Scala
case class Config(_config: String)

abstract class Foo(val config: Config) {
  def this(config: String) = {
    this(Config(config))
  }
}

class Bar(config: String) extends Foo(config) {
  def foo(): Unit = {
    config.getClass()
  }
}

new Bar("").foo() //This line will crash in prod
```

## Output

```
Caused by: java.lang.ClassCastException: class Playground$Config cannot be cast to class java.lang.String (Playground$Config is in unnamed module of loader sbt.internal.BottomClassLoader @11415fc3; java.lang.String is in module java.base of loader 'bootstrap')
at Playground$Bar.config$accessor(main.scala:11)
	at Playground$Bar.foo(main.scala:13)
	at Playground$.&lt;clinit&gt;(main.scala:17)
```

## Expectation
That code compiles and runs without crashing (like in scala 2.13)

--- a/compiler/src/dotty/tools/dotc/transform/PostTyper.scala
+++ b/compiler/src/dotty/tools/dotc/transform/PostTyper.scala
@@ -116,7 +116,8 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
      *  This info is used in phase ParamForwarding
      */
     private def forwardParamAccessors(impl: Template)(using Context): Unit = impl.parents match
-      case superCall @ Apply(fn, superArgs) :: _ if superArgs.nonEmpty =>
+      case superCall @ Apply(fn, superArgs) :: _
+      if superArgs.nonEmpty && fn.symbol.isPrimaryConstructor =>
         fn.tpe.widen match
           case MethodType(superParamNames) =>
             for case stat: ValDef <- impl.body do

--- a/tests/run/i19711.scala
+++ b/tests/run/i19711.scala
@@ -1,0 +1,29 @@
+class Foo(val s: Any):
+  def this(s: String) =
+    this(0)
+class Bar(s: String) extends Foo(s):
+  def foo = s
+
+class Foo2(val s: Any)
+class Bar2(s: String) extends Foo2(s):
+  def foo = s
+
+case class Config(_config: String)
+
+abstract class Foo3(val config: Config) {
+  def this(config: String) = {
+    this(Config(config))
+  }
+}
+
+class Bar3(config: String) extends Foo3(config) {
+  def foo(): Unit = {
+    config.getClass()
+  }
+}
+
+
+@main def Test =
+  Bar("").foo
+  Bar2("").foo
+  Bar3("").foo()

